docs/how-to-guides/performance_analysis.md

```diff
@@ -0,0 +1,210 @@
+- Isolate the paired core if hyper-threading enabled
+
+To fulfill these conditions on Linux, a custom kernel build with the following kernel configurations is required.
+Note that even if Full tickless is enabled, timer interrupts are generated for scheduling if more than two tasks exist in one core.
+
+```text
+# Enable CONFIG_NO_HZ_FULL
+-> General setup
+-> Timers subsystem
+-> Timer tick handling (Full dynticks system (tickless))
+(X) Full dynticks system (tickless)
+
+# Allows RCU callback processing to be offloaded from selected CPUs
+# (CONFIG_RCU_NOCB_CPU=y)
+-> General setup
+-> RCU Subsystem
+-*- Offload RCU callback processing from boot-selected CPUs
+```
+
+Additionally, the kernel boot parameters need to be set as follows.
+
+```text
+GRUB_CMDLINE_LINUX_DEFAULT=
+  "... isolcpus=2,8 rcu_nocbs=2,8 rcu_nocb_poll nohz_full=2,8 intel_pstate=disable”
+```
+
+In the above configuration, for example, the node to be measured is assumed to run on core 2, and core 8, which is a hyper-threading pair, is also being isolated.
+Appropriate decisions on which cores to run the measurement target and which nodes to isolate need to be made based on the cache and core layout of the measurement machine.
+You can easily check if it is properly configured by running `cat /proc/softirqs`.
+Since `intel_pstate=disable` is specified in the kernel boot parameter, `userspace` can be specified in the scaling governor.
+
+```shell
+cat /sys/devices/system/cpu/cpu2/cpufreq/scaling_governor // ondemand
+sudo sh -c "echo userspace > /sys/devices/system/cpu/cpu2/cpufreq/scaling_governor"
+```
+
+This allows you to freely set the desired frequency within a defined range.
+
+```shell
+sudo sh -c "echo <freq(kz)> > /sys/devices/system/cpu/cpu2/cpufreq/scaling_setspeed"
+```
+
+Turbo Boost needs to be switched off on Intel CPUs, which is often overlooked.
+
+```shell
+sudo sh -c "echo 0 > /sys/devices/system/cpu/cpufreq/boost"
+```
+
+### Run Single Node Separately
+
+Following the instructions in the `ros2_single_node_replayer` README, start the node and play the dedicated rosbag created by the tool.
+Before playing the rosbag, appropriately set the CPU affinity of the thread on which the node runs, so it is placed on the isolated core prepared.
+
+```shell
+taskset --cpu-list -p <target cpu> <pid>
+```
+
+To avoid interference in the last level cache, minimize the number of other applications running during the measurement.
+
+### Measurement and Visualization
+
+To visualize the performance of the measurement target, embed code for logging timestamps and performance counter values in the target source code.
+To achieve this objective, a tool called `pmu_analyzer` has been prepared.
+
+[sykwer/pmu_analyzer](https://github.com/sykwer/pmu_analyzer)
+
+Details on how to use the tool can be found in the README.
+This tool can measure the turnaround time of any section in the source code, as well as various performance counters.
+
+## Case Studies
```

Comment:
I think it would be better if you have specific commit/repos file that can reproduce your case study. (e.g., pcl::fromROSMsg can be removed in the future)

Reply:
Thank you for your review.
We have added a link for the pull request so that everyone can reference the modifications even after the PCL is removed.
https://github.com/autowarefoundation/autoware-documentation/pull/339/commits/9301d671866981c6b0fc09fad270f4dd63b5de0d

Reply:
Thanks for the fix. However, it is a bit different from what I expected. I was expecting to know the commits to reproduce your graphs shown in this page.

Reply:
I also feel a bit strange that you start your case studies with a conclusion.

For example, for sensing case study, you start with saying that you should reduce the number of copies of pointcloud, but shouldn't that be concluded from your performance analysis, not the other way around? 

I would like to know your intention why you would like to add these case studies. If it just there to give tips about coding, then it might be okay as is, but if you put it to explain how these performance analysis should be done, then I think we should change how it is written.